### PR TITLE
add: book catalogue page, bookPreview  "see more" part

### PR DESCRIPTION
**Pull request by sabrina34352**
image_2022-06-09_14-56-38
image_2022-06-09_14-56-51

### CODE REVIEW

File: packages/frontapp/src/app/components/AllBooksList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+export interface IMaterial {
+  id: string;
+  __typename?: 'Material';
+  picture?: string | null | undefined;
+  category: string;
+  author: string;
+  title: string;
+  created_at: string;
+}
```
Comment (x8er) [inline packages/frontapp/src/app/components/AllBooksList/index.tsx:22]: You could just export this type from global-types.ts

File: packages/frontapp/src/app/components/CategoriesList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import styled from '@emotion/styled';
+import { colors, dimensions } from '@mimir/ui-kit';
+
+interface IStyledContainerProps {
+  flex: string | null;
+}
+
+interface IProps {
+  allCategories: { [category: string]: number } | undefined;
+}
```
Comment (x8er) [inline packages/frontapp/src/app/components/CategoriesList/index.tsx:11]: ICategoriesListProps. Even if you don't export this interface

File: packages/frontapp/src/app/components/AllBooksList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+interface IProps {
+  items: Array<IMaterial | null> | undefined;
+  sortingCategory: string | undefined;
+}
```
Comment (x8er) [inline packages/frontapp/src/app/components/AllBooksList/index.tsx:27]: IAllBooksListProps

File: packages/frontapp/src/app/pages/BookPreview.tsx [hunk 4]
```diff
@@ -18,7 +22,7 @@ const ButtonWrapper = styled.div`
   align-items: center;
   cursor: pointer;
 `;
-const ButtonGRoup = styled.div`
+export const ButtonGRoup = styled.div`
```
Comment (x8er) [inline packages/frontapp/src/app/pages/BookPreview.tsx:25]: ButtonGroup

**Review by x8er — CHANGES_REQUESTED** [review]

File: packages/frontapp/src/app/components/AllBooksList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                category={item?.category}
+                date={item?.created_at}
+              />
+            )
+        )}
+    </WrapperList>
+  );
+};
+
+export default index;
```
Comment (x8er) [inline packages/frontapp/src/app/components/AllBooksList/index.tsx:50]: See how we call component constants

**Review by x8er — CHANGES_REQUESTED** [review]

File: packages/frontapp/src/app/components/AllBooksList/index.tsx (hunk 1 above)
Comment (sabrina34352) [reply packages/frontapp/src/app/components/AllBooksList/index.tsx:22]: it doesnt work, and requires a lot of additional attributes..

Comment (x8er) [reply packages/frontapp/src/app/components/AllBooksList/index.tsx:22]: @sabrina34352 Sorry, from client-types.ts in apollo-client lib. https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys

Comment (sabrina34352) [reply packages/frontapp/src/app/components/AllBooksList/index.tsx]: @x8er yes i did it, thank you for pointing out

**Review by x8er — CHANGES_REQUESTED** [review]

**Review by x8er — APPROVED** [review]

**Review by IvanUglovets — APPROVED** [review]

**Pull request merged**